buildbot/master/files/config/steps.yml

```diff
@@ -22,6 +22,12 @@ mac-rel-css:
   - bash ./etc/ci/lockfile_changed.sh
   - bash ./etc/ci/manifest_changed.sh
 
+mac-nightly:
+  - ./mach build --release
+  - ./mach package --release
+  - s3cmd -c ~/.s3cfg-servo put target/release/*.tar.gz s3://servo-developer-preview/nightly/`date +%Y`/
+  - rm -rf target/
```

Comment:
Why did you add the `rm -rf` lines? Buildbot will clean out the working directory at the start of each build IIRC (or I think maybe the `git` step does this), so it should not be necessary to have as a standalone step.